From de8ab715cc2af65df377c6ab67cb8231d368bba0 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:37:16 -0400
Subject: [virt] kvm/vmx: Use proper types to access const arrays

Message-id: <1406040016-3289-42-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86120
O-Subject: [RHEL7 PATCH v2 041/221] KVM: VMX: Use proper types to access const arrays
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mathias Krause <minipli@googlemail.com>

Use a const pointer type instead of casting away the const qualifier
from const arrays. Keep the pointer array on the stack, nonetheless.
Making it static just increases the object size.

Signed-off-by: Mathias Krause <minipli@googlemail.com>
Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit c2bae8939487c101b0516c4c8ad80f543b22bb29)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index d87fbd4..d2c4b0c 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -5982,8 +5982,8 @@ static void copy_shadow_to_vmcs12(struct vcpu_vmx *vmx)
  unsigned long field;
  u64 field_value;
  struct vmcs *shadow_vmcs = vmx->nested.current_shadow_vmcs;
- unsigned long *fields = (unsigned long *)shadow_read_write_fields;
- int num_fields = max_shadow_read_write_fields;
+ const unsigned long *fields = shadow_read_write_fields;
+ const int num_fields = max_shadow_read_write_fields;
 
  vmcs_load(shadow_vmcs);
 
@@ -6012,12 +6012,11 @@ static void copy_shadow_to_vmcs12(struct vcpu_vmx *vmx)
 
 static void copy_vmcs12_to_shadow(struct vcpu_vmx *vmx)
 {
- unsigned long *fields[] = {
-  (unsigned long *)shadow_read_write_fields,
-  (unsigned long *)shadow_read_only_fields
+ const unsigned long *fields[] = {
+  shadow_read_write_fields,
+  shadow_read_only_fields
  };
- int num_lists =  ARRAY_SIZE(fields);
- int max_fields[] = {
+ const int max_fields[] = {
   max_shadow_read_write_fields,
   max_shadow_read_only_fields
  };
@@ -6028,7 +6027,7 @@ static void copy_vmcs12_to_shadow(struct vcpu_vmx *vmx)
 
  vmcs_load(shadow_vmcs);
 
- for (q = 0; q < num_lists; q++) {
+ for (q = 0; q < ARRAY_SIZE(fields); q++) {
   for (i = 0; i < max_fields[q]; i++) {
    field = fields[q][i];
    vmcs12_read_any(&vmx->vcpu, field, &field_value);
-- 
1.7.1